Data layer new protocol added

jeffh1996 opened the pull request:
Added the new protocol in the data layer but kept the old protocol (Will need to make a new PR(s) to change to the new protocol since many parts of the program will be affected and I do not want to make this PR too large).

Note: Since CMUs are non-existent anymore, this might also need to be part of a new PR since a lot of files are affected.

enochtsang commented on src/DataLayer/BatteryData.h:
```diff
@@ -95,4 +95,85 @@ struct BatteryData
+    unsigned short int averageCellVoltage;
+    PrechargeState prechargeState;
+    const char* prechargeStateJSON;
+    unsigned char auxVoltage;
```
Why is it all just added commented code, why isn't it actually changed?

enochtsang commented on src/DataLayer/CcsDefines.h:
```diff
@@ -10,12 +10,12 @@ namespace CcsDefines
 
     const int MPPT_COUNT = 3;
-    const int CMU_COUNT = 4;
+    const int CMU_COUNT = 4;				//TODO:Delete all references to CMUs
```
Not TODO, it's part of this PR.

jeffh1996 commented:
Will reopen after making some changes

enochtsang commented on src/DataLayer/CcsDefines.h:
```diff
@@ -30,4 +29,7 @@ namespace CcsDefines
     const unsigned int MAJOR_MC_ERROR_MASK = 0xFF;
+
+    //No data
+    const bool NO_DATA = 0;
```
I like it, but maybe leave a comment on why this exists. The "No data" comment isn't helpful at all.

jeffh1996 commented:
Added a description for the NO_DATA member and added some data checks in the tests to ensure that the data members before and after the NO_DATA members are correct.

As for the CMU files. They should have already been removed in the previous PR and when I updated this branch in 2eafa6abf5aef63209b6d1e56d78500194c73b4e .
